### DISCUSSION TITLE
Introducing ASReview-Preprocess extension

### DISCUSSION THREAD

**Discussion by rohitgarud** [asreview/asreview#1406] [Show and tell]
Hello all,
I am working on the [ASReview-preprocess](https://github.com/rohitgarud/asreview-preprocess) extension for catering to all preprocessing needs such as finding missing data for records and deduplication. The purpose of this extension is to make the research accessible to those who cannot access paid resources and tools such as Endnote.

The current functionality includes CLI for:
- Update:
    1. Find missing DOIs using title and year with [CrossRef API](https://www.crossref.org/documentation/retrieve-metadata/rest-api/)
    2. Find missing metadata including abstracts using [OpenAlex API](https://docs.openalex.org/) with the help of [pyalex]() library developed by @J535D165 
    3. Save retrieved metadata from OpenAlex to the local database to avoid burdening OpenAlex with repeated requests for the same record and use saved record metadata in future
    4. Python API provides ways to add other APIs for retrieving missing information and using different local databases.
- Dedup:
   1. Use authors, title, abstract, year, journal, doi, volume, pages, number (issue) and ISBN for deduplicating the records using fuzzy string matching. The [recordlinkage library](https://github.com/J535D165/recordlinkage) developed by none other than @J535D165 is used for implementing data matching for deduplication
   2. Currently default deduplication available with [ASReview-datatools](https://github.com/asreview/asreview-datatools) and the default deduplication in Endnote software is available
   3. Python API provides functions for developing other deduplication strategies along with functions for unifying (cleaning pipeline) different fields for better deduplication

Working on and looking for collaborators/contributors for:
- Webapp
- Other deduplication strategies like [Bramer Method](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4915647/) used with Endnote and [ASySD](https://www.biorxiv.org/content/10.1101/2021.05.04.442412v1) (R package) ([ASySD](https://github.com/camaradesuk/ASySD) has greatly influenced the development of this extension)
- Add other tools from ASReview-datatools such as compose and stack with GUI 
- Test coverage
- Documentation

Please check the [repository](https://github.com/rohitgarud/asreview-preprocess) and give suggestions in the discussion section of the repository. PRs are very much welcome. 
Note: The extension is under development and is not ready to be used in the ASReview workflow just yet.

**Comment by Rensvandeschoot** [comment]
I'm glad to see that you're focusing on data quality in the ASReview-preprocess extension. As a statistician, I couldn't agree more with the importance of high-quality data when it comes to active learning for text screening. The famous statement "garbage in, garbage out" holds true, and I always encourage high-quality data input for high-quality results.

The ASReview-Datatools repository contains various data preprocessing tools, including a simple deduplication script based on DOI, title, and abstract (https://github.com/asreview/asreview-datatools). However, more advanced deduplication strategies could undoubtedly be beneficial.

In addition, retrieving missing data is crucial for data quality. We apply a similar strategy for the benchmark datasets (https://github.com/asreview/systematic-review-datasets), and a new release incorporating this strategy is coming soon.

Your work on the ASReview-preprocess extension is highly appreciated, and it's great to see that you're following similar paths. Have you considered merging your developments into the Datatools package? We're open to discussing the possibility of integrating your work or understanding if there are reasons for keeping the projects separate. Parallel development is always an option, and we're eager to collaborate and support each other's efforts.

Looking forward to hearing your thoughts!!

**Reply by rohitgarud** [reply]
Thank you @Rensvandeschoot for your feedback. The plan is to use features from datatools and provide a GUI using Webapp for different preprocessing steps. Please go through the issues of ASReview-Preprocess repository, there I have highlighted a few additional planned features. If the features can be easily merged with Datatools, then I am happy to do so because many users are already familiar with Datatools. I would like to get the opinion of ASReview team @J535D165 @terrymyc @PeterLombaers @jspaaks @jteijema @gimoAI and others from the development point of view.
